src/components/BatchClassComponents/ExerciseBatchClass/ExerciseBatchClass.js

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useState } from "react";
+import useMediaQuery from "@mui/material/useMediaQuery";
+import { Container, Grid, Typography } from "@mui/material";
+import { Link } from "react-router-dom";
+import { CardMedia, CardContent, Card, Button, Stack } from "@mui/material";
+import { Box } from "@mui/system";
+import axios from "axios";
+import useStyles from "../styles";
+import { useSelector } from "react-redux";
+import { breakpoints } from "../../../theme/constant";
+import { TimeLeft } from "../../../constant";
+import { format } from "../../../common/date";
+import RevisionClassEnroll from "../Revision/RevisionClassEnroll";
+// import { Button } from "framework7-react";
+import DropOut from "../DropOut";
+import ClassJoinTimerButton from "../../Class/ClassJoinTimerButton";
+const ExerciseBatchClass = (props) => {
+  const isActive = useMediaQuery("(max-width:" + breakpoints.values.sm + "px)");
+  const classes = useStyles();
+  const [dropOutOpen, setDropOutOpen] = useState(false);
+  const { facilitator, start_time, end_time, is_enrolled, meet_link, id } =
+    props;
+  const closeDropOut = () => {
+    setDropOutOpen(false);
+  };
+  let [TimeLefts, setTimeLefts] = useState(TimeLeft(start_time));
+  var ONE_MINUTE = 60 * 1000;
+  setInterval(() => {
+    setTimeLefts(TimeLeft(start_time));
+  }, ONE_MINUTE);
+  return TimeLefts !== "expired" ? (
```

Comment:
We'll want to use some sort of Timer component for this as well.  Also, there generally isn't a good reason to use `var` instead of `let` (or `const` as appropriate) anymore.  And this creates a new repeating timer *every* time that the Component is rendered and these timers don't go away until you leave merakilearn.org.  So we'll want to fix this as well.

Reply:
Will be resolved by https://github.com/navgurukul/bhanwari-devi/pull/488